book/course/theory.md

```diff
@@ -0,0 +1,141 @@
+# Theory
+
+## Introduction
+
+### Why
+
+This lesson provides supporting information to the tools used to use, build and maintain software on the Universities HPC systems by introducing you to the terminology and technologies that will be covered by further lessons in this course.
+
+### Overview of topics
+
+* Linux ????
+* Build Tools
+* Package Managers
+* Platforms
+
+### Types of files (WHERE DOES THIS FIT ???????????????)
+
+There are numerous file types you need to be aware of when using and developing code for HPC.
+Below are the key types of files:
+
+* executables - these types of files contain compiled binary code. When opened, the operating system will execute the instructions of the code to run the program.
+* Libraries - these files usually contain compiled code, but cannot be directly run (they can also contain data, templates, documentation). They provide interfaces to be accessed from an existing running program. Libraries are useful for sharing common functionality between different programs.
```

Comment:
Not sure I like this definition, although I think the additional note that the term often gets used to cover other things is valid.  I think in terms of building software, I'd go with more of a:

A Library in Linux
A library is a collection of pre-compiled pieces of code called functions. The library contains common functions and together, they form a package called — a library.

Or maybe:

In programming, a library is an assortment of pre-compiled pieces of code that can be reused in a program.